### ISSUE TITLE
Image renders with strange white "scars"

### ISSUE THREAD

**Issue by simplenotezy** [dompdf/dompdf#920]
I have this QR code:

![image](https://cloud.githubusercontent.com/assets/3786627/6732295/1fe4b44c-ce4b-11e4-9cf0-c258c0a6bbd6.png)

That is generated using this library: http://phpqrcode.sourceforge.net/

I embed it in my HTML using this line:

```
ob_start();
\PHPQRCode\QRcode::png('contents of the QR code', null, 'H', 8, 2);
$QRCode = ob_get_contents();
ob_end_clean();

echo '<img src="data:image/png;base64,' . base64_encode($QRCode) . '" />';
```

However, when I stream it to the browser, it looks weird. Each stream generates different looking QR codes, like this:

![image](https://cloud.githubusercontent.com/assets/3786627/6732361/a2af69e4-ce4b-11e4-8513-43edca96726e.png)

![image](https://cloud.githubusercontent.com/assets/3786627/6732366/abc20c08-ce4b-11e4-9ba3-57d83856297f.png)

![image](https://cloud.githubusercontent.com/assets/3786627/6732420/0401cfac-ce4c-11e4-8f4e-820d236ce95c.png)

I have embeded other images in the PDF that renders just fine.

**Comment by bsweeney** [comment]
I'm not seeing the same issue (using the image you attached). Which version of dompdf are you using? PHP? GD? Do you have either the gmagick or imagick extension loaded?

**Comment by canfiax** [comment]
This is my composer:

    "dompdf/dompdf" : "dev-master#cf7d8a0a27270418850cc7d7ea532159e5eeb3eb",
    "aferrandini/phpqrcode": "1.0.1"

GD Setup:
![image](https://cloud.githubusercontent.com/assets/3786627/6764805/2b184e68-cfc3-11e4-8eb0-ba663bb639fd.png)

![image](https://cloud.githubusercontent.com/assets/3786627/6764806/31c15c14-cfc3-11e4-94d7-ffc8691b2c0f.png)

PDF backend option:

    def("DOMPDF_PDF_BACKEND", "CPDF");

The issue only appears on my production server - things seem to run smoothly on localhost.

**Comment by bsweeney** [comment]
I imagine that viewing the HTML you feed into dompdf does not indicate the same issue. What about if you view the loaded document using `$dompdf->output_html()`?

I have the same release of ImageMagick and don't see the issue, so I don't know if it's related. But you could test that by disabling the imagick module in your php.ini or commenting out the relevant code in dompdf/lib/class.pdf.php (lines 3927-3993 plus line 4038).

**Comment by Cracken6** [comment]
I had same issue with corrupted images (white dots and lines in image).
Disabling Imagick fixed this issue. Thanks!

**Comment by jammin84** [comment]
I can also confirm, removing ImagicK resolved it for me. not sure why.

Version I had installed was in the ubuntu trusty repository.
php5-imagick                        3.1.2-1build1

Before + After
![image](https://cloud.githubusercontent.com/assets/6802951/16183414/1d9a979e-36e4-11e6-8b60-fd98986824a2.png)

**Comment by bsweeney** [comment]
Merged a fix for this in commit c68484c. If anyone can confirm the issue is resolved that would be great. The fix will be released shortly in 0.8.2.
